crates/compiler/gen_dev/src/generic64/x86_64.rs

```diff
@@ -1955,6 +1955,24 @@ impl Assembler<X86_64GeneralReg, X86_64FloatReg> for X86_64Assembler {
         andpd_freg64_freg64(buf, dst, src);
     }
 
+    #[inline(always)]
+    fn abs_freg32_freg32(
+        buf: &mut Vec<'_, u8>,
+        relocs: &mut Vec<'_, Relocation>,
+        dst: X86_64FloatReg,
+        src: X86_64FloatReg,
+    ) {
+        movss_freg32_rip_offset32(buf, dst, 0);
+
+        // TODO: make sure this constant only loads once instead of every call to abs
+        relocs.push(Relocation::LocalData {
+            offset: buf.len() as u64 - 4,
+            data: 0x7fffffffu64.to_le_bytes().to_vec(),
+        });
+
+        andps_freg32_freg32(buf, dst, src);
```

Comment:
That is not what I expected, but makes sense.

Reply:
this is what godbolt told me, so :man_shrugging: